cli/tools/jupyter/comm.rs

```diff
@@ -0,0 +1,147 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+
+use deno_core::error::AnyError;
+use ring::hmac;
+use zeromq::prelude::*;
+use zeromq::util::PeerIdentity;
+use zeromq::SocketOptions;
+
+use super::hmac_verify;
+use super::ReplyMessage;
+use super::RequestMessage;
+use super::SideEffectMessage;
+
+pub struct PubComm {
+  conn_str: String,
+  hmac_key: hmac::Key,
+  socket: zeromq::PubSocket,
+
+  // TODO(bartlomieju):
+  #[allow(unused)]
+  identity: String,
```

Comment:
Generally, identities aren't strictly necessary in ZeroMQ for a regular Jupyter setup since it relies on a single server that intermediates between a web client and the kernel. It's probably good to keep the identity filed here for compatibility and future-proofing. IIRC, Spyder had support for multiple simultaneous connections to a single kernel (on the server side). Now people just open more websockets to the server itself for separate connections.